zerver/lib/actions.py

```diff
@@ -3451,3 +3453,56 @@ def do_set_name_changes_disabled(realm, name_changes_disabled):
         value=name_changes_disabled,
     )
     send_event(event, active_user_ids(realm))
+
+def notify_realm_custom_profile_fields(realm):
+    # type: (Realm) -> None
+    fields = custom_profile_fields_for_realm(realm.id)
+    event = dict(type="custom_profile_fields", fields=fields)
+    send_event(event, active_user_ids(realm))
+
+def do_add_realm_custom_profile_field(realm, name, field_type):
+    # type: (Realm, Text, int) -> int
+    with transaction.atomic():
+        custom_field = CustomProfileField(realm=realm, name=name,
+                                          field_type=field_type)
+        custom_field.full_clean()
+        custom_field.save()
+        realm.profile_fields.append(custom_field.as_dict())
+        realm.save(update_fields=['profile_fields'])
+
+    notify_realm_custom_profile_fields(realm)
+    return custom_field.id
+
+def do_remove_realm_custom_profile_field(realm, field_id):
+    # type: (Realm, int) -> None
+    with transaction.atomic():
+        field_id = int(field_id)
+        CustomProfileField.objects.filter(realm=realm, pk=field_id).delete()
+        index = realm.get_profile_field_index(field_id)
+        realm.profile_fields.pop(index)
+        realm.save(update_fields=['profile_fields'])
```

Comment:
This should at least get a comment that it's going to delete the CustomProfileFieldValue data via a cascade, and possibly should be more explicit than that (do you need to pass the `cascade` argument for this to work?  Worth adding a test for deleting a field with data attached).